src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,134 @@
+                            this._data.addTask(index, td); // since index is -1 of size
+                            System.out.println(this._ui.showTaskAdded(td, tasks));
+                        }
+                        break;
+                    case "deadline":
+                        if (cmdLine.isEmpty()) {
+                            throw new DeadlineException(cmdLine, 3);
+                        } else if (!cmdLine.contains("/by")) {
+                            throw new DeadlineException(cmdLine, 1);
+                        } else {
+                            String[] cmdSplit = cmdLine.split("/by ");
+                            if (cmdSplit.length <= 1) {
+                                throw new DeadlineException(cmdLine, 3);
+                            } else if (cmdSplit[0].equals(" ")) {
+                                throw new DeadlineException(cmdLine, 2);
+                            } else {
+                                Deadline dl = new Deadline(cmdSplit[0], cmdSplit[1]);
+                                int index = tasks.getSize() + 1;
+                                this._data.addTask(index, dl); // since index is -1 of size
+                                System.out.println(this._ui.showTaskAdded(dl, tasks));
+                            }
+                        }
+                        break;
+                    case "event":
+                        if (cmdLine.isEmpty()) {
+                            throw new EventException(cmdLine, 3);
+                        } else if (!cmdLine.contains("/at")) {
+                            throw new EventException(cmdLine, 1);
+                        } else {
+                            String[] cmdSplitt = cmdLine.split("/at ");
+                            if (cmdSplitt.length <= 1) {
+                                throw new EventException(cmdLine, 3);
+                            } else if (cmdSplitt[0].equals(" ")) {
+                                throw new EventException(cmdLine, 2);
+                            } else {
+                                Event ev = new Event(cmdSplitt[0], cmdSplitt[1]);
+                                int index = tasks.getSize() + 1;
+                                this._data.addTask(index, ev); // since index is -1 of size
+                                System.out.println(this._ui.showTaskAdded(ev, tasks));
+                            }
+                        }
+                        break;
+                    case "find":
+                        System.out.println(this._ui.showFind(cmdLine, this._data.load()));
+                        break;
+                    default: // if it is not any of the above commands
+                        throw new DukeException(cmdWord + cmdLine);
+                }
+            } catch (DukeException e) {
+                System.err.println(Ui.addLines(e.getMessage()));
+            } catch (ToDoException e) {
```

Comment:
There is no need to have one Exception class for every Task.
As much as possible, you should use standard Java exceptions and only use the custom DukeException when absolutely necessary. See https://github.com/nus-cs2103-AY1920S1/forum/issues/20